provider/aws.go

```diff
@@ -236,20 +233,23 @@ func (p *AWSProvider) submitChanges(changes []*route53.Change) error {
 // changesByZone separates a multi-zone change into a single change per zone.
 func changesByZone(zones map[string]*route53.HostedZone, changeSet []*route53.Change) map[string][]*route53.Change {
 	changes := make(map[string][]*route53.Change)
+	zoneNameIDMapper := map[string]string{}
 
 	for _, z := range zones {
+		zoneNameIDMapper[aws.StringValue(z.Name)] = aws.StringValue(z.Id)
 		changes[aws.StringValue(z.Id)] = []*route53.Change{}
 	}
 
 	for _, c := range changeSet {
 		hostname := ensureTrailingDot(aws.StringValue(c.ResourceRecordSet.Name))
 
-		zone := suitableZone(hostname, zones)
-		if zone == nil {
+		zone := zoneFinder(hostname, zoneNameIDMapper)
```

Comment:
`zoneFinder` could actually return the ID in the first place. Then we wouldn't need the mapping stuff here. Or it returns both `zoneId, zoneName` so the caller can pick what it wants.